### DISCUSSION TITLE
Save Image and Preview Image combined node with passthru/masking

### DISCUSSION THREAD

**Discussion by ricperry** [Comfy-Org/ComfyUI#2945] [Ideas]
The number of times I need a Preview Image or a Save Image node and switch between them is a little out of control. Add to that the fact that sometimes I need to edit a mask on one of the previews, and it all adds up to:

Can we get a more flexible Save Image node? I'd recommend it replace both the Save Image and Preview Image nodes. There should be a (true/false) toggle to actually save the image. There should be 2 outputs: IMAGE, MASK. Mask editor already works on the node. This would reduce the number of noodles outputting from a VAE Decode when you're going to send the image on to somewhere else. 

And if we're REALLY going to make things efficient, this could also replace the Load Image node.  But that might be a step too far??

**Comment by citkane** [comment]
I have been looking for this myself and found some helpful nodes in the [ComfyUI-Impact-Pack](https://github.com/ltdrdata/ComfyUI-Impact-Pack) custom node bundle. There are a lot of nodes in it, but once installed have a look at the `PreviewBridge (image)` and `PreviewBridge (latent)` nodes in the menu: "Add Node / ImpactPack / Util / ..."

**Reply by ltdrdata** [reply]
FYI, I am providing tutorials for my nodes via separated repo.

https://github.com/ltdrdata/ComfyUI-extension-tutorials/blob/Main/ComfyUI-Impact-Pack/tutorial/previewbridge.md
https://github.com/ltdrdata/ComfyUI-extension-tutorials?tab=readme-ov-file

And video tutorial as well.
(Currently, most are non-verbal, and I am gradually replacing them with versions that have some narration.)
https://www.youtube.com/watch?v=O3EjONCS-rU
